### PR TITLE
Don’t return an error if stored hash is not found for a sequence

### PR DESCRIPTION
**Pull request by ajres**
fixes couchbaselabs/sync-gateway-accel/issues/127

### CODE REVIEW

File: db/sequence_hasher.go [hunk 1]
```diff
@@ -258,14 +258,14 @@ func (s *sequenceHasher) GetClock(sequence string) (*base.SequenceClockImpl, err
 		return clock, loadErr
 	}
 
+	clock = base.NewSequenceClockImpl()
 	if uint16(len(storedClocks.Sequences)) <= seqHash.collisionIndex {
-		base.LogTo("Changes+", "Stored hash not found for sequence [%s] collision index [%d], #storedClocks:%d", sequence, seqHash.collisionIndex, len(storedClocks.Sequences))
-		return clock, errors.New(fmt.Sprintf("Stored hash not found for sequence [%s], returning zero clock", sequence))
+		base.LogTo("Changes+", "Stored hash not found for sequence [%s] collision index [%d], #storedClocks:%d, returning zero clock", sequence, seqHash.collisionIndex, len(storedClocks.Sequences))
```
Comment (adamcfraser) [inline db/sequence_hasher.go]: If we're not returning any information to the caller, this should be logging a warning (and not just a log key that might not be enabled).

**Comment by adamcfraser** [comment]
Can you provide a bit of context on why handling this in the sequence hasher is the preferred approach?

File: db/sequence_hasher.go (hunk 1 above)
Comment (ajres) [reply db/sequence_hasher.go]: Changed to base.Warn

**Comment by ajres** [comment]
The two main call paths into sequenceHasher.GetClock() are 

These direct calls 

* db.parseClockSequenceID
    * DatabaseContext.ParseSequenceID
        * handler.HandleChanges
        * handler.updateChangesOptionsFromQuery
    * SequenceID.unmarshalClockSequence
        * SequenceID.UnmarshalJSON
             
For the direct call paths from the changes feed handler, returning a new zero sequence clock would result in additional change entries on the feed, but no error.  

The indirect calls from JSON.Unmarshall to the custom SequenceID.UnmarshalJSON are harder to analyse. But through code inspection it does not appear that returning a zero clock in these cases will cause errors in SG. To validate that no issues are introduced with CBL clients would probably best be identified by running the functional tests against this change.

* db.ParseSequenceIDFromJSON
* rest.readChangesOptionsFromJSON
* db.ChangesOptions (struct Since value)
* db.ChangeEntry (struct Seq value)

**Pull request merged**